templates/CRM/Group/Form/Search.tpl

```diff
@@ -164,6 +165,10 @@
                 .prepend('{/literal}<span class="collapsed show-children" title="{ts}show child groups{/ts}"/></span>{literal}')
                 .find('div').css({'display': 'inline'});
             });
+            $('tbody tr.crm-smart-group', settings.nTable).each(function () {
+              $(this).find('td:first')
+                .append(smartGroupText);
```

Comment:
One thing is that the parentsOnly var is only set on first page load, not when the table is refreshed by ajax, so changing any of the search filters at the top of the table like Group Type or Visibility then refreshes the table, but doesn't apply this text.

Reply:
Hi @demeritcowboy thanks for your feedback. I have updated my patch and also checked with parent group. Can you please review my patch again?

Reply:
Thanks it's better, except now it adds "Smart Group" to _every_ child group even if they are not smart kids. Hmm this seems harder than it looks.